Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -96,24 +96,28 @@ protected override void ProcessRecord()
             switch (ParameterSetName)
             {
                 case ParameterSetNames.List:
-                    DatasetsResource.ListRequest lRequest = new DatasetsResource.ListRequest(Service, Project);
+                    DatasetsResource.ListRequest lRequest = Service.Datasets.List(Project);
                     lRequest.All = All;
                     lRequest.Filter = Filter;
-                    var lResponse = lRequest.Execute();
-                    if (lResponse != null) { 
-                        WriteObject(lResponse, true);
-                    }
-                    else
+                    do
                     {
-                        WriteError(new ErrorRecord(
-                            new Exception("400"), 
-                            "Error 400: List request to server failed.",
-                            ErrorCategory.InvalidArgument, 
-                            Project));
+                        DatasetList response = lRequest.Execute();
+                        if (response == null)
+                        {
+                            WriteError(new ErrorRecord(
+                                new Exception("List request to server responded with null."),
+                                "List request returned null", ErrorCategory.InvalidArgument, Project));
+                        }
+                        if (response.Datasets != null)
+                        {
+                            WriteObject(response.Datasets, true);
+                        }
+                        lRequest.PageToken = response.NextPageToken;
                     }
+                    while (!Stopping && lRequest.PageToken != null);
                     break;
                 case ParameterSetNames.Get:
-                    DatasetsResource.GetRequest gRequest = new DatasetsResource.GetRequest(Service, Project, Dataset);
+                    DatasetsResource.GetRequest gRequest = Service.Datasets.Get(Project, Dataset);
```

Comment:
Adding a function here as well will help readability. The two case statements would look similar and short, making them easier to understand together.